disnake/guild.py

```diff
@@ -4553,3 +4566,468 @@ async def create_automod_rule(
+    ) -> PlaceholderID:
+        """
+        Adds a category channel to the guild builder.
+
+        There is an alias for this named ``add_category_channel``.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The category's name.
+        overwrites: Dict[``PlaceholderID``, :class:`PermissionOverwrite`]
+            A :class:`dict` of roles to :class:`PermissionOverwrite`\\s which can be synced to channels.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            A placeholder ID for the created category.
+        """
+        _id, _ = self._add_channel(type=ChannelType.category, name=name, overwrites=overwrites)
+        return _id
+
+    add_category_channel = add_category
+
+    def add_text_channel(
+        self,
+        name: str,
+        *,
+        overwrites: Dict[PlaceholderID, PermissionOverwrite] = MISSING,
+        category: PlaceholderID = MISSING,
+        topic: Optional[str] = MISSING,
+        slowmode_delay: int = MISSING,
+        nsfw: bool = MISSING,
+        default_auto_archive_duration: AnyThreadArchiveDuration = MISSING,
+    ) -> PlaceholderID:
+        """
+        Adds a text channel to the guild builder.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The channel's name.
+        overwrites: Dict[``PlaceholderID``, :class:`PermissionOverwrite`]
+            A :class:`dict` of roles to :class:`PermissionOverwrite`\\s to apply to the channel.
+        category: ``PlaceholderID``
+            The category to place the new channel under.
+
+            .. warning::
+                Unlike :func:`Guild.create_text_channel`, the parent category's
+                permissions will *not* be synced to this new channel by default.
+
+        topic: Optional[:class:`str`]
+            The channel's topic.
+        slowmode_delay: :class:`int`
+            Specifies the slowmode rate limit for users in this channel, in seconds.
+            A value of ``0`` disables slowmode. The maximum value possible is ``21600``.
+            If not provided, slowmode is disabled.
+        nsfw: :class:`bool`
+            Whether to mark the channel as NSFW or not.
+        default_auto_archive_duration: Union[:class:`int`, :class:`ThreadArchiveDuration`]
+            The default auto archive duration in minutes for threads created in this channel.
+            Must be one of ``60``, ``1440``, ``4320``, or ``10080``.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            A placeholder ID for the created text channel.
+        """
+        _id, data = self._add_channel(
+            type=ChannelType.text,
+            name=name,
+            overwrites=overwrites,
+            category=category,
+            topic=topic,
+            slowmode_delay=slowmode_delay,
+            nsfw=nsfw,
+        )
+
+        if default_auto_archive_duration is not MISSING:
+            data["default_auto_archive_duration"] = cast(
```

Comment:
Why is this here and not in `_add_channel`?

Reply:
text channels are the only ones that support this field at the moment, the fields implemented in `_add_channel` are generally there to avoid duplicating code

Reply:
Makes sense 👍

Reply:
you can mark this as resolved